Assets/Scripts/MazeGenerator.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System.Collections;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MazeGenerator : MonoBehaviour
+{
+    [SerializeField]
+    private int width;
+
+    [SerializeField]
+    private int height;
+
+    public Transform brick;
+    private int[,] Maze;
+
+    private enum TileType { path, wall };
+    private List<Vector3> pathMazes = new List<Vector3>();
+    private Stack<Vector2> tileStack = new Stack<Vector2>();
+
+    private readonly List<Vector2> offsets = new List<Vector2>
+    {
+        new Vector2(0, 1),
+        new Vector2(0, -1),
+        new Vector2(1, 0),
+        new Vector2(-1, 0)
+    };
+
+    private System.Random rnd = new System.Random();
+
+    private Vector2 currentTile;
+    public Vector2 CurrentTile
+    {
+        get { return currentTile; }
+
+        private set
+        {
+            if (value.x < 1 || value.x >= this.width - 1 || value.y < 1 || value.y >= this.height - 1)
+            {
+                throw new ArgumentException("CurrentTile must be within the one tile border all around the maze");
+            }
+            else if (value.x % 2 == 1 || value.y % 2 == 1)
+            {
+                currentTile = value;
+            }
+            else
+            {
+                throw new ArgumentException("The current square must not be both on an even X-axis and an even Y-axis, to ensure we can get walls around all tunnels");
+            }
+        }
+    }
+
+    private static MazeGenerator instance;
+    public static MazeGenerator Instance
+    {
+        get { return instance; }
+    }
+
+    private void Awake()
+    {
+        instance = this;
+        GenerateMaze();
+    }
+
+
+    private void GenerateMaze()
+    {
+        Maze = new int[width, height];
+        for (int x = 0; x < width; x++)
+        {
+            for (int y = 0; y < height; y++)
+            {
+                Maze[x, y] = (int)TileType.wall;
+            }
+        }
+
+        CurrentTile = Vector2.one;
+        tileStack.Push(CurrentTile);
+        PrimsAlg();
+
+        for (int i = 0; i <= Maze.GetUpperBound(0); i++)
+        {
+            for (int j = 0; j <= Maze.GetUpperBound(1); j++)
+            {
+                if (Maze[i, j] == (int)TileType.wall)
+                {
+                    Instantiate(brick, new Vector3(i * brick.localScale.x, 0, j * brick.localScale.z), Quaternion.identity);
+                }
+                else if (Maze[i, j] == (int)TileType.path)
+                {
+                    pathMazes.Add(new Vector3(i, 0, j));
+                }
+            }
+        }
+    }
+
+    private void PrimsAlg()
```

Comment:
This is actually an implementation of prims algorithm. It is a generation method based on it, but certainly not it. I can go over why with you if you want. But I would name it `RandomMazeGenerator` and then provide a comment that is based on prims algorithm but does not use weights to inform generation and instead randomly chooses neighbors